pkg/provisioner/controller.go

```diff
@@ -115,8 +117,16 @@ func (c *obcController) Start(stopCh <-chan struct{}) error {
 	if !cache.WaitForCacheSync(stopCh, c.obcHasSynced, c.obHasSynced) {
 		return fmt.Errorf("failed to waith for caches to sync ")
 	}
-	go wait.Until(c.runWorker, time.Second, stopCh)
-
+	var count int
+	threadiness, set := os.LookupEnv("LIB_BUCKET_PROVISIONER_THREADS")
```

Comment:
It is more typical to see ENV variables collected in _main_ and then passed as args to other functions. If we did this then every provisioner that wanted >1 thread would have to either get the LIB_BUCKET_PROVISIONER_THREADS env variable or hard-code their value (with 1 being the lib's default). @copejon @guymguym any thoughts on this?

Reply:
The library has now `main` function, but rather just the Run() method.  

> If we did this then every provisioner that wanted >1 thread would have to either get the LIB_BUCKET_PROVISIONER_THREADS

Not sure I follow.  The env var is defined and accessed by the library.  There's no need for dependees to also have to define/get it.  Rather, it's just something that would be optionally set in the runtime environment.  If not set, defaults to 1.

Reply:
I was wondering (out loud) if the lib should _not_ get the LIB_BUCKET_PROVISIONER_THREADS env variable. Instead, each provisioner passes to the lib the # of threads they want with the lib default remaining 1. Then each provisioner decides how to determine their own # of threads.

Reply:
If the library defines an env var, but then requires that dependees get the value and pass it to the constructor rather than checking it itself, then the env var doesn't actually have any purpose.  Since the value is passed through the constructor function, where it's set is then completely arbitrary.

This also defeats the purpose of using an env var, which was to not make changes to the library interface.  So if we're defining an env var (which I still think we should), then the library should access it directly.

Reply:
1) I was thinking the lib would _not_ define an env var,
2) `defeats the purpose of using an env var, which was to not make changes to the library interface` I completely agree and retract my comment.